src/test/java/seedu/address/storage/GemCommandStorageTest.java

```diff
@@ -0,0 +1,120 @@
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.model.ReadOnlyGradPad;
+import seedu.address.model.module.Module;
+
+public class GemCommandStorageTest {
+    public static final String TEST_FOUNDATION_PATH = "src/main/resources/data/foundationmodules.json";
+    public static final String TEST_SCIENCE_PATH = "src/main/resources/data/sciencemodules.json";
+    private GemCommandStorage storage = new GemCommandStorage();
+    private ObservableList<Module> requiredFoundation;
+    public void setUpRequiredFoundation() throws IOException, DataConversionException {
+        JsonGradPadStorage storage = new JsonGradPadStorage(Paths.get(TEST_FOUNDATION_PATH));
+        ReadOnlyGradPad gradPad = storage.readGradPad().get();
+        requiredFoundation = gradPad.getModuleList();
+    }
+    @Test
+    public void getGehModules_validTest() {
+        ObservableList<Module> actual = storage.getGehModules();
+        assertNull(actual);
+    }
+    @Test
+    public void getGeqModules_validTest() {
+        ObservableList<Module> actual = storage.getGeqModules();
+        assertNull(actual);
+    }
+    @Test
+    public void getGerModules_validTest() {
+        ObservableList<Module> actual = storage.getGerModules();
+        assertNull(actual);
+    }
+    @Test
+    public void getGesModules_validTest() {
+        ObservableList<Module> actual = storage.getGesModules();
+        assertNull(actual);
+    }
+    @Test
+    public void getGetModules_validTest() {
+        ObservableList<Module> actual = storage.getGetModules();
+        assertNull(actual);
+    }
+    @Test
+    public void setGehModulesInvalidPath_throwsIOexception() {
+        assertThrows(IOException.class, () -> storage.setGehModules(INVALID_PATH));
+    }
+    @Test
+    public void setGeqModulesInvalidPath_throwsIOexception() {
+        assertThrows(IOException.class, () -> storage.setGeqModules(INVALID_PATH));
+    }
+    @Test
+    public void setGerModulesInvalidPath_throwsIOexception() {
+        assertThrows(IOException.class, () -> storage.setGerModules(INVALID_PATH));
+    }
+    @Test
+    public void setGesModulesInvalidPath_throwsIOexception() {
+        assertThrows(IOException.class, () -> storage.setGesModules(INVALID_PATH));
+    }
+    @Test
+    public void setGetModulesInvalidPath_throwsIOexception() {
+        assertThrows(IOException.class, () -> storage.setGetModules(INVALID_PATH));
+    }
+    @Test
+    public void setGehModulesValidPath_success() throws IOException,
+            DataConversionException, IllegalValueException {
+        setUpRequiredFoundation();
+        ObservableList<Module> actual = requiredFoundation;
+        storage.setGehModules(FOUNDATION_PATH);
+        ObservableList<Module> expected = storage.getGehModules();
```

Comment:
I think you mixed up `actual` and `expected` HAHA 🤣  in the other tests below too

Reply:
rip HAHAH changed accordingly